acme: Allow custom ACME server directory URL

jannispinter opened the pull request:
The underlying `acme.sh` allows custom ACME server URLs (using `--server`). Adding the necessary fields to specify a custom ACME server URL from UCI.

Maintainer: @tohojo 
Compile tested: ipq40xx
Run tested: ipq40xx (tested issuance of a certificate from a custom ACME compatible CA)

Signed-off-by: Jannis Pinter <jannis+openwrt@pinterjann.is>

tohojo commented on net/acme/files/run.sh:
```diff
@@ -210,11 +210,14 @@ issue_cert()
 	local ret
 	local domain_dir
+	local use_custom_acme_directory
+	local custom_acme_directory_url
```
Why two variables? Also, please use shorter variable names...

jannispinter replied:
Thanks, I will update the PR as requested. 

I have changes for this feature ready to push for the corresponding LuCI application, and added `use_custom_acme_directory` for a checkbox that conditionally shows/hides the input field for the URL (I don't want to show the input field, when the user doesn't need/understand this option).

tohojo reviewed (changes requested):
Fine with adding this but please fix the variable names. Also, you need to bump the package revision, and fix you S-o-B as the bot says :)

jannispinter commented:
I have added the Signed-off-by line, bumped `PKG_RELEASE` and changed the variable names. I decided to remove `use_custom_acme_directory` entirely, as it is indeed redundant. When the checkbox in LuCI is unchecked, the `directory_url` won't get written to UCI.

tohojo commented on net/acme/files/run.sh:
```diff
@@ -210,6 +210,7 @@ issue_cert()
 	local user_cleanup
 	local ret
 	local domain_dir
+	local directory_url
```
What does "directory_url" mean? Why not just "acme_server" or "server_url"?

jannispinter replied:
The resource an ACME client connects to is called "Directory" (see [RFC 8555](https://tools.ietf.org/html/rfc8555#page-23)). Most implementations put that resource at `/directory` (such as Let's Encrypt, which uses the following directory URL: https://acme-v02.api.letsencrypt.org**/directory**)

I have implemented both client and server versions of the RFC and might have a bit of a tunnel vision on the naming here. I think changing it to `acme_server` actually makes more sense and is a very concise name.

tohojo reviewed (changes requested):
Cool, that's better. Going to bikeshed the variable name a bit more, though ;)

tohojo reviewed (approved):
Great, that's better :)

Going to assume those test failures are unrelated, so let's just merge this!
